lib/useAuth.js

```diff
@@ -20,6 +24,14 @@ export async function signUp(email, password, data) {
     await updateProfile(user, data);
     sendEmailVerification(user);
 }
+export const signUpwithGoogle = async () => {
+    try {
+        await signInWithPopup(auth, googleProvider);
+        Router.push("/events");
+    } catch (err) {
+        console.error(err);
```

Comment:
Very good job @SufyanCS! There is one little suggestion I have:

Instead of try catching or console erroring the error here, what you can do is returning just returning `signInWithPopup()` ,like the `signUp()` function, and then in the signin or signup page you can handle that error with the try catch while using the `AuthErrorBox` to show that error to the user.

Check the signup function in the signup page for more reference!

Reply:
Done, Check and review the code again.